File: sdk/src/main/java/org/p2p/solanaj/kits/transaction/TransactionDetails.java
package org.p2p.solanaj.kits.transaction;

public abstract class TransactionDetails {

    private final String signature;
    private final long blockTime;
    private final int slot;

    public TransactionDetails(String signature, long blockTime, int slot) {
        this.signature = signature;
        this.blockTime = blockTime;
        this.slot = slot;
    }

    public abstract TransactionDetailsType getType();

    public abstract Object getInfo();

    public String getData() {
        return null;
    }

    public String getSignature() {
        return signature;
    }

    public int getSlot() {
        return slot;
    }

    public long getBlockTime() {
        /*
         * Since blocktime is time of when the transaction was processed in SECONDS
         * we are converting it into milliseconds
         * */
        return blockTime * 1000;
    }
}

File: solanaj/src/main/java/org/p2p/solanaj/kits/transaction/TransactionDetails.kt
package org.p2p.solanaj.kits.transaction

import org.p2p.solanaj.model.types.ConfirmationStatus
import java.util.concurrent.TimeUnit

sealed class TransactionDetails(
    val signature: String,
    val blockTimeSeconds: Long,
    val slot: Int,
    var error: List<Any>? = null,
    var status: ConfirmationStatus? = null
) {
    abstract val type: TransactionDetailsType

    /*
    * Since blocktime is time of when the transaction was processed in SECONDS
    * we are converting it into milliseconds
    * */
    val blockTimeMillis: Long
        get() = TimeUnit.SECONDS.toMillis(blockTimeSeconds)
}
